gwpy/table/io/sql.py

```diff
@@ -93,7 +93,8 @@ def fetch(engine, tablename, columns=None, selection=None, **kwargs):
     # Convert unicode columns to string
     types = tab.apply(lambda x: pd.api.types.infer_dtype(x.values))
 
-    for col in types[types == 'unicode'].index:
-        tab[col] = tab[col].astype(str)
+    if not tab.empty:
```

Comment:
why is this new clause required?

Reply:
Because if the query is valid but the resulting table is empty then that loop fails as it never knows what `type` each column is because no columns where filled in.